lib/smart_proxy_dns_powerdns/backend/mysql.rb

```diff
@@ -35,15 +35,20 @@ def create_record domain_id, name, type, content
       name = connection.escape(name)
       content = connection.escape(content)
       type = connection.escape(type)
-      connection.query("INSERT INTO records (domain_id, name, ttl, content, type) VALUES (#{domain_id}, '#{name}', #{ttl.to_i}, '#{content}', '#{type}')")
+      connection.query("INSERT INTO records (domain_id, name, ttl, content, type, change_date) VALUES (#{domain_id}, '#{name}', #{ttl.to_i}, '#{content}', '#{type}', UNIX_TIMESTAMP())")
       connection.affected_rows == 1
     end
 
     def delete_record domain_id, name, type
       name = connection.escape(name)
       type = connection.escape(type)
       connection.query("DELETE FROM records WHERE domain_id=#{domain_id} AND name='#{name}' AND type='#{type}'")
-      connection.affected_rows >= 1
+      ret = connection.affected_rows >= 1
+      if ret
+        connection.query("UPDATE records SET change_date=UNIX_TIMESTAMP() WHERE domain_id=#{domain_id} AND type='SOA'")
+        ret = connection.affected_rows == 1
```

Comment:
> This is not true for MySQL. An UPDATE statement will set affected_rows to the number of rows that changed. If the column already contains the value then it will not increase the affected_rows. Since unix timestamps have a resolution of 1 second there's a decent chance this will happen.

This is really bad news :( We will have to discard the return value of this call then because there is a high chance of this happening when Foreman deletes multiple hosts of the same domain.
Also I am wondering if PowerDNS can serve a zone without SOA record anyway so the problem can be reduced to the "already up to date" race stated above.

Should I discard the return value of the UPDATE statement then?